src/containers/tud/components/ProgressBar.js

```diff
@@ -0,0 +1,146 @@
+// @flow
+
+import React, { useEffect, useState } from 'react';
+
+import styled from 'styled-components';
+import { Colors, Typography } from 'lattice-ui-kit';
+import { DateTime } from 'luxon';
+
+const { PURPLES, NEUTRALS, ORANGE } = Colors;
+
+const Grid = styled.div`
+  align-items: center;
+  display: grid;
+  grid-template-columns: ${(props) => `${props.completedRatio}fr auto`};
+`;
+
+const Wrapper = styled.div`
+  background: white;
+  margin-bottom: 10px;
+  padding: 20px 0;
+  position: sticky;
+  top: 0;
+  z-index: 10;
+`;
+
+const ProgressIndicator = styled.div`
+  border-top-color: ${(props) => props.color};
+  border-top-style: solid;
+  border-top-width: 5px;
+`;
+
+const ProgressLabelWrapper = styled.div`
+  align-items: center;
+  display: flex;
+  justify-content: space-between;
+  padding-top: 5px;
+
+  :last-of-type {
+    padding-left: 10px;
+  }
+`;
+
+const validateDateTimes = (dateTimes :Array<?DateTime>) => {
+  const isValid = dateTimes.map((dateTime) => dateTime && dateTime.isValid);
+  return isValid.every((value) => value);
+};
+
+type Props = {
+  currentTime :?DateTime;
+  dayEndTime :?DateTime;
+  dayStartTime :?DateTime;
+  is12hourFormat :boolean;
+  isDayActivityPage :boolean;
+}
+
+const ProgressBar = (props :Props) => {
+
+  const {
+    currentTime,
+    dayEndTime,
+    dayStartTime,
+    is12hourFormat,
+    isDayActivityPage,
+  } = props;
+
+  const [completedRatio, setCompletedRatio] = useState(0);
+  const [isOutOfRange, setIsOutOfRange] = useState(true);
+
+  useEffect(() => {
+    if (validateDateTimes([dayStartTime, dayEndTime, currentTime])) {
+      // $FlowFixMe
+      const totalTimeDiff :Object = dayEndTime.diff(dayStartTime).toObject().milliseconds;
+      // $FlowFixMe
+      const completed :Object = currentTime.diff(dayStartTime).toObject().milliseconds;
+
+      let ratio = [
+        completed / totalTimeDiff,
+        (totalTimeDiff - completed) / totalTimeDiff
+      ];
+
+      const isValid = ratio.every((val) => val >= 0);
+      setIsOutOfRange(isValid);
+
+      if (ratio[0] < 0) ratio = [0, 1];
+      if (ratio[1] < 0) ratio = [1, 0];
+
+      setCompletedRatio(ratio[0]);
+    }
+  }, [dayStartTime, dayEndTime, currentTime]);
+
+  if (!isDayActivityPage) {
+    return null;
+  }
+
+  const formatTime = (input :?DateTime) => {
+    if (!validateDateTimes([input])) return null;
+
+    return is12hourFormat
+      // $FlowFixMe
```

Comment:
What's the error?

Reply:
property `toLocaleString` missing on null/undefined.

however, `validateDateTimes()` call pre-checks if the value is null/undefined. Therefore, `toLocaleString` can never be called on invalid date values,  but flow doesn't seem to be aware.